tests/test_code_conventions.py

```diff
@@ -48,26 +48,38 @@ def test_spdx():
         print(result)
         assert False
 
-# Ensure "free" is not used unprotected in the main OQS code.
-@helpers.filtered_test
-@pytest.mark.skipif(sys.platform.startswith("win"), reason="Not needed on Windows")
-def test_free():
+def test_memory_functions():
     c_files = []
     for path, _, files in os.walk('src'):
-        c_files += [os.path.join(path,f) for f in files if f[-2:] == '.c']
+        c_files += [os.path.join(path, f) for f in files if f.endswith('.c')]
+
+    memory_functions = ['free', 'malloc', 'calloc', 'realloc', 'strdup']
     okay = True
+
     for fn in c_files:
         with open(fn) as f:
-            # Find all lines that contain 'free(' but not '_free('
-            for no, line in enumerate(f,1):
-                if not re.match(r'^.*[^_]free\(.*$', line):
+            content = f.read()
+            lines = content.splitlines()
+            for no, line in enumerate(lines, 1):
+                # Skip comments
+                if line.strip().startswith('//') or line.strip().startswith('/*'):
                     continue
-                if 'IGNORE free-check' in line:
+                # Check if we're inside a multi-line comment
+                if '/*' in content[:content.find(line)] and '*/' not in content[:content.find(line)]:
```

Comment:
Just a note, this will produce a false positive in the following scenario:
```c
/* comment */
/*
malloc(sizeof(int));
*/
OQS_MEM_malloc(sizeof(int));
```
I think you'd have to search backwards for the closest `/*`, but that seems like overkill.

Reply:
@SWilson4 You are right. We can improve it later.

Reply:
> We can improve it later.

@songlingatpan Such statements tend to get lost after a merge: Can I suggest creating an issue documenting this as an area for improvement, then? I'd really hate having to deal with problems that we saw coming but didn't fix beforehand.

Reply:
@baentsch I will add some code to handle it shortly.